nixos/modules/services/networking/whoogle-search.nix

```diff
@@ -0,0 +1,124 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.whoogle-search;
+  whoogle-search = pkgs.whoogle-search;
+  stateDir = "/var/lib/whoogle-search";
+in
+{
+  options = {
+    services.whoogle-search = {
+
+      enable = mkEnableOption "Whether to enable whoogle-search service.";
+
+      user = mkOption {
+        type = types.str;
+        default = "whoogle-search";
+        description = "The user as which to run whoogle-search daemon.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "whoogle-search";
+        description = "The group as which to run whoogle-search daemon.";
+      };
+
+      port = mkOption {
+        type = types.port;
+        default = 5000;
+        description = ''
+          Port to listen on.
+        '';
+      };
+
+      listenAddress = mkOption {
+        type = types.str;
+        default = "127.0.0.1";
+        description = ''
+          Address to listen on for the web interface.
+        '';
+      };
+
+      dataDir = mkOption {
+        type = types.path;
+        default = stateDir;
+        description = ''
+          The directory where whoogle-search will create session files.
+        '';
+      };
+
+      httpsOnly = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Enforce HTTPS redirects for all requests.
+        '';
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    users.users = mkIf (cfg.user == "whoogle-search") {
+      whoogle-search = {
+        group = cfg.group;
+        uid = config.ids.uids.whoogle-search;
```

Comment:
As mentioned above you could try to replace the user with `DynamicUser`. Alternatively, set `isSystemUser` and avoid setting `uid` and `gid`.

See https://github.com/NixOS/rfcs/blob/master/rfcs/0052-dynamic-ids.md for details and explanation.

Reply:
`DynamicUser` options seems to work well. I have removed the user and group creation. Thanks!